enumerables.rb

```diff
@@ -0,0 +1,137 @@
+# rubocop: disable Metrics/ModuleLength, Style/CaseEquality, Style/IfInsideElse
+
+module Enumerable
+  def my_each(var = nil)
+    return to_enum unless block_given?
+
+    n = if var.nil?
+          0
+        else
+          var
+        end
+    while n <= size - 1
+      yield(to_a[n])
+      n += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n], n)
+      n += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each do |n|
+      result.push(n) if yield(n)
+    end
+    result
+  end
+
+  def my_all?(var = nil)
+    result = true
+    my_each do |n|
+      if block_given?
+        result = false unless yield(n)
+      elsif var.nil?
+        result = false unless n
+      else
+        result = false unless var === n
+      end
+    end
+    result
+  end
+
+  def my_any?(var = nil)
+    result = false
+    my_each do |n|
+      if block_given?
+        return result = true if yield(n)
+      elsif var.nil?
+        return result = true if n
+      else
+        return result = true if var === n
+      end
+    end
+    result
+  end
+
+  def my_none?(var = nil)
+    result = true
+    my_each do |n|
+      if block_given?
+        return result = false if yield(n)
+      elsif var.nil?
+        return result = false if n
+      else
+        return result = false if var === n
+      end
+    end
+    result
+  end
+
+  def my_count(var = nil)
+    result = []
+    if block_given?
+      my_each do |n|
+        result.push(n) if yield(n)
+      end
+    elsif var.nil?
+      return length
+    else
+      my_each do |n|
+        result.push(n) if n == var
+      end
+    end
+    result.length
+  end
+
+  def my_map(var = nil)
+    result = []
+    return to_enum unless block_given? && var.nil?
+
+    if var
+      my_each do |n|
+        result.push(var.call(n))
+      end
+    else
+      my_each do |n|
+        result.push(yield(n))
+      end
+    end
+    result
+  end
+
+  def my_inject(var = nil)
+    if var
+      if var.is_a?(Symbol)
+        result = self[0]
+        my_each(1) do |n|
+          result = result.method(var).call(n)
+        end
+        return result
+      else
+        result = var
+      end
+    elsif var.nil?
+      result = self[0]
+    end
+    my_each(1) do |n|
+      result = yield(result, n)
+    end
+    result
+  end
```

Comment:
- [ ] should have working `my_inject`  identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(5..50).my_inject(4) { |prod, n| prod * n } == (5..50).inject(4) { |prod, n| prod * n }  => 5069015533618896340602101361010794807396273594826752000000000000 
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/78707916-84831280-78ce-11ea-970e-56f8ed6040cc.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/78707946-91a00180-78ce-11ea-99e8-d401a87e5766.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
(5..50).my_inject(:+) == (5..50).inject(:+) => 1265
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/78708077-c744ea80-78ce-11ea-9ac0-85b774a21431.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/78708114-d3c94300-78ce-11ea-9255-55002a684a20.png)

Reply:
Made the changes, it seems like it's working now!